src/lib/serviceRoutinesV2/getEntities.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+*
+* Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Ken Zangelin
+*/
+#include <string>
+#include <vector>
+
+#include "rest/ConnectionInfo.h"
+#include "ngsi/ParseData.h"
+#include "apiTypesV2/Entities.h"
+#include "rest/EntityTypeInfo.h"
+#include "serviceRoutinesV2/getEntities.h"
+#include "serviceRoutines/postQueryContext.h"
+
+
+
+/* ****************************************************************************
+*
+* getEntities - 
+*
+* GET /v2/entities
+*
+* Payload In:  None
+* Payload Out: Entities
+*
+* URI parameters:
+*   - limit=NUMBER
+*   - offset=NUMBER
+*   - count=true/false
+*
+* 01. Fill in QueryContextRequest
+* 02. Call standard op postQueryContext
+* 03. Render Entities response
+* 04. Cleanup and return result
+*/
+std::string getEntities
+(
+  ConnectionInfo*            ciP,
+  int                        components,
+  std::vector<std::string>&  compV,
+  ParseData*                 parseDataP
+)
+{
+  std::string  answer;
+  Entities     entities;
+
+
+  // 01. Fill in QueryContextRequest
+  parseDataP->qcr.res.fill(".*", "", "true", EntityTypeEmptyOrNotEmpty, "");
```

Comment:
You will see this in more than one place ...
It was implemented like that initially but as the payload comes in as ASCII it is a lot easier to work this way. The code got full of 'string to bool conversions' and got difficult to read.